### PR TITLE
Fix GitGitGadget on Linux and with different Git versions

### PR DESCRIPTION
**Pull request by dscho**
This patch series addresses the assumptions in the current `master` branch that apply only to my own Windows setup, but probably not to anybody else's. And certainly not to the VSTS build which I want to use as "release gate", i.e. I want to require that build to pass before merging any Pull Request.

### CODE REVIEW

**Comment by dscho** [comment]
@derrickstolee any chance you can review this?

File: lib/patch-series.ts [hunk 1]
```diff
@@ -68,7 +70,9 @@ export class PatchSeries {
                 }
             });
 
-            rangeDiff = await git(["range-diff", "--no-color", range]);
+            if (await gitCommandExists("range-diff", workDir)) {
```
Comment (derrickstolee) [inline lib/patch-series.ts]: What happens if `rangeDiff` stays uninitialized?

Comment (derrickstolee) [reply lib/patch-series.ts]: (I see higher up that it is initialized to "", so things probably aren't _broken_, but not great)

Comment (dscho) [reply lib/patch-series.ts]: If it is empty, or false, or undefined, nothing gets inserted in `insertRangeDiff()`: https://github.com/dscho/gitgitgadget/blob/2fb305a285398b8ac957ef35f51e0be5e295c63a/lib/patch-series.ts#L356-L358

**Pull request merged**